### PR TITLE
Update eventlistener doc to include eventlistener responsibility

### PR DESCRIPTION
**Pull request by savitaashture**
# Changes

Added information about how Deployment and Service created by Eventlistener

Fix: https://github.com/tektoncd/triggers/issues/699

Signed-off-by: Savita Ashture sashture@redhat.com

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/triggers/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```
None
```

### CODE REVIEW

File: docs/eventlisteners.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ using [Event Interceptors](#Interceptors).
     - [Interceptors](#interceptors)
 - [Logging](#logging)
 - [Labels](#labels)
+- [EventListener Liabilities](#eventlistener-liabilities)
```
Comment (dibyom) [inline docs/eventlisteners.md:25]: Maybe we just call this section "How does the EventListener work?"

Comment (savitaashture) [reply docs/eventlisteners.md]: Done

File: docs/eventlisteners.md [hunk 2]
```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
 - `namespace` - Refers to the namespace of the EventListener
 - `eventID` - Refers to the uniqueID that gets assigned to each incoming request
 
+## How does the EventListener work?
+
+By now its clear that what are the configuration file for an `EventListener` with `Required` and `Optional` fields explanation.
+
+Lets understand how EventListener works with sample Github example
+
+Create a sample GitHub example
```
Comment (dibyom) [inline docs/eventlisteners.md:725]:
```suggestion
* Create a sample GitHub example
```

File: docs/eventlisteners.md [hunk 3]
```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
...
+
+```bash
+kubectl create -f https://github.com/tektoncd/triggers/tree/master/examples/github
+```
+Once creation is success we see a new `Deployment` and `Service` running with the help of `kubectl` command line tool
```
Comment (dibyom) [inline docs/eventlisteners.md:730]:
```suggestion
* Once the EventListener is created,  the Triggers controller will create a new `Deployment` and `Service` for the EventListener. We can use `kubectl` to see them running:
```

File: docs/eventlisteners.md [hunk 4]
```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
...
+```bash
+kubectl get deployment
+NAME                             READY   UP-TO-DATE   AVAILABLE   AGE
+el-github-listener-interceptor   1/1     1            1           11s
+
+kubectl get svc
+NAME                             TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
+el-github-listener-interceptor   ClusterIP   10.99.188.140   <none>        8080/TCP   52s
+```
+It indicates that EventListener code take all information from eventlistener yaml file which are listed in [Syntax](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#syntax), [Lables](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#labels)
```
Comment (dibyom) [inline docs/eventlisteners.md:740]:
```suggestion
The Triggers controller uses fields from the EventListener's `spec` (which is described in the [Syntax](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#syntax) section, as well as [`metadata.labels`](https://github.com/tektoncd/triggers/blob/master/docs/eventlisteners.md#labels)
```

File: docs/eventlisteners.md [hunk 5]
```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
...
+and also hardcode some of the information like `Image, Name, Port` etc... to create **Deployment** and **Service**, So its the responsibility of eventlistener to create Deployment and Service.
```
Comment (dibyom) [inline docs/eventlisteners.md:741]:
```suggestion
in addition to some pre-configured information like (container `Image`, `Name`, `Port`) to create the **Deployment** and **Service**. ```

Comment (dibyom) [inline docs/eventlisteners.md:741]: Probably worth adding:

We follow a naming convention when creating these resources. An EventListener named `foo` will create a deployment and a service both named `el-foo`.

File: docs/eventlisteners.md [hunk 6]
```diff
@@ -715,6 +716,39 @@ The EventListener responds with following message after receiving the event:
...
+
+Once all the resources are up and running user can get URL and access it.
```
Comment (dibyom) [inline docs/eventlisteners.md:743]:
```suggestion
Once all the resources are up and running user can get a URL to send webhook events to. This URL points to the service created above and points to the deployment.
```

**Review by dibyom — COMMENTED** [review]
Thanks so much for writing this up and apologies for the delay in reviewing. Made a few small suggestions but overall looks good!

**Comment by savitaashture** [comment]
> Thanks so much for writing this up and apologies for the delay in reviewing. Made a few small suggestions but overall looks good!

Thanks a lot @dibyom 

Addressed all the review comments

**Review by khrm — COMMENTED** [review]
/approve

**Review by vdemeester — COMMENTED** [review]
/lgtm

**Pull request merged**